x-pack/filebeat/input/websocket/input_test.go

```diff
@@ -399,6 +401,164 @@ var inputTests = []struct {
+			"url": "ws://testapi/getresults",
+			"input_initializer_program": `
+			{
+				"url" : (
+					has(state.cursor) && has(state.cursor.since) ? 
+						state.url+"?since="+ state.cursor.since 
+					: 	
+						state.url+"?since="+ state.start_time
+					)
+			}`,
+			"state": map[string]interface{}{
+				"initial_start_time": "2022-01-01T00:00:00Z",
+			},
+		},
+		wantErr: fmt.Errorf("failed eval: ERROR: <input>:4:51: no such key: start_time"),
+	},
+	{
+		name: "missing cursor variable",
+		config: map[string]interface{}{
+			"url": "ws://testapi/getresults",
+			"input_initializer_program": `
+			{
+				"url" : (
+					has(state.cursor) ? 
+						state.url+"?since="+ state.cursor.since 
+					: 	
+						state.url+"?since="+ state.initial_start_time
+					)
+			}`,
+			"state": map[string]interface{}{
+				"cursor": map[string]interface{}{
+					"timestamp": "2017-08-17T14:54:12",
+				},
+				"initial_start_time": "2022-01-01T00:00:00Z",
+			},
+		},
+		wantErr: fmt.Errorf("failed eval: ERROR: <input>:4:24: no such key: since"),
+	},
+	{
+		name: "missing curly braces in program definition",
+		config: map[string]interface{}{
+			"url": "ws://testapi/getresults",
+			"input_initializer_program": `
+				"url" : (
+					has(state.cursor) && has(state.cursor.since) ?
+						state.url+"?since="+ state.cursor.since
+					:
+						state.url+"?since="+ state.initial_start_time
+					)`,
+			"state": map[string]interface{}{
+				"cursor": map[string]interface{}{
+					"since": "2017-08-17T14:54:12",
+				},
+				"initial_start_time": "2022-01-01T00:00:00Z",
+			},
+		},
+		wantErr: fmt.Errorf("failed compilation: ERROR: <input>:2:11: Syntax error: mismatched input ':' expecting <EOF>"),
+	},
+}
+
+func TestInputInitializer(t *testing.T) {
+	// tests will ignore context cancelled errors, since they are expected
+	ctxCancelledError := fmt.Errorf("context canceled")
+	logp.TestingSetup()
+	for _, test := range inputInitializerTests {
+		t.Run(test.name, func(t *testing.T) {
+
+			cfg := conf.MustNewConfigFrom(test.config)
+
+			conf := config{}
+			conf.Redact = &redact{}
+			err := cfg.Unpack(&conf)
+			if err != nil {
+				if test.wantErr != nil {
+					assert.EqualError(t, err, test.wantErr.Error())
+					return
+				}
+				t.Fatalf("unexpected error unpacking config: %v", err)
+			}
+
+			name := input{}.Name()
+			if name != "websocket" {
+				t.Errorf(`unexpected input name: got:%q want:"websocket"`, name)
+			}
+
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer cancel()
+
+			var state map[string]interface{}
+			if conf.State == nil {
+				state = make(map[string]interface{})
+			} else {
+				state = conf.State
+			}
+
+			response, err := input{test.time, conf}.initializeInputURL(ctx, state, logp.NewLogger("websocket_input_initializer_test"))
+			if (fmt.Sprint(err) != fmt.Sprint(ctxCancelledError)) && (strings.Split(fmt.Sprint(err), "\n")[0] != fmt.Sprint(test.wantErr)) {
```

Comment:
```suggestion
			if !errors.Is(err, context.Canceled) && strings.Split(fmt.Sprint(err), "\n")[0] != fmt.Sprint(test.wantErr) {
```
but I'm not convinced that we need syntax error checking here. This would be akin to ensuring that addition works, these behaviours are already tested in the library.